### PR TITLE
Deprecate 'jar-path' in spring cloud

### PR DESCRIPTION
**Pull request by metacpp**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

---
This change is [Reviewable](https://reviewable.io/reviews/azure/azure-cli-extensions/2395)

### CODE REVIEW

**Comment by yonzhan** [comment]
spring cloud

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls help to review? thanks.

File: src/spring-cloud/azext_spring_cloud/_help.py [hunk 1]
```diff
@@ -142,7 +142,7 @@
     - name: Deploy source code to an app. This will pack current directory, build binary with Pivotal Build Service and then deploy to the app.
       text: az spring-cloud app deploy -n MyApp -s MyCluster -g MyResourceGroup
     - name: Deploy a pre-built jar to an app with jvm options and environment variables.
-      text: az spring-cloud app deploy -n MyApp -s MyCluster -g MyResourceGroup --jar-path app.jar --jvm-options="-XX:+UseG1GC -XX:+UseStringDeduplication" --env foo=bar
+      text: az spring-cloud app deploy -n MyApp -s MyCluster -g MyResourceGroup --artifact-path app.jar --jvm-options="-XX:+UseG1GC -XX:+UseStringDeduplication" --env foo=bar
```
Comment (yungezz) [inline src/spring-cloud/azext_spring_cloud/_help.py:145]: is --jar-path a ga parameter? If yes, pls follow the guideline to deprecate it https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments. Remove a GAed param will break existing users.

File: src/spring-cloud/azext_spring_cloud/_help.py [hunk 2]
```diff
@@ -244,7 +244,7 @@
     - name: Deploy source code to a new deployment of an app. This will pack current directory, build binary with Pivotal Build Service and then deploy.
       text: az spring-cloud app deployment create -n green-deployment --app MyApp -s MyCluster -g MyResourceGroup
     - name: Deploy a pre-built jar to an app with jvm options and environment variables.
-      text: az spring-cloud app deployment create -n green-deployment --app MyApp -s MyCluster -g MyResourceGroup --jar-path app.jar --jvm-options="-XX:+UseG1GC -XX:+UseStringDeduplication" --env foo=bar
+      text: az spring-cloud app deployment create -n green-deployment --app MyApp -s MyCluster -g MyResourceGroup --artifact-path app.jar --jvm-options="-XX:+UseG1GC -XX:+UseStringDeduplication" --env foo=bar
```
Comment (yungezz) [inline src/spring-cloud/azext_spring_cloud/_help.py:247]: if there test covering this param? if yes, pls re-record test also.

File: src/spring-cloud/azext_spring_cloud/_help.py (hunk 1 above)
Comment (metacpp) [reply src/spring-cloud/azext_spring_cloud/_help.py:145]: That's right, and we'd like to deprecate it. I've followed the guideline to redirect it. See below message:

```
Option '--jar-path' has been deprecated and will be removed in a future release. Use '--artifact-path' instead.
This command usually takes minutes to run. Add '--verbose' parameter if needed.
file_type is NetCoreZip
[1/3] Requesting for upload URL
[2/3] Uploading package to blob
[3/3] Updating deployment in app 'planet-weather-provider' (this operation can take a while to complete)
```

File: src/spring-cloud/azext_spring_cloud/_help.py (hunk 2 above)
Comment (metacpp) [reply src/spring-cloud/azext_spring_cloud/_help.py:247]: The test will still use the old/new arguments side by side to keep the backward compatibility. In next minor release, we will remove the old argument in the test case to completely.

**Pull request closed without merging**